src/libstd/collections/hash/map.rs

```diff
@@ -1547,45 +1590,152 @@ impl<'a, K, V> Entry<'a, K, V> {
 
 impl<'a, K, V> OccupiedEntry<'a, K, V> {
     /// Gets a reference to the key in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::HashMap;
+    ///
+    /// let mut map: HashMap<&str, usize> = HashMap::new();
+    /// map.entry("poneyland").or_insert(12);
+    /// assert_eq!(map.entry("poneyland").key(), &"poneyland");
+    /// ```
     #[stable(feature = "map_entry_keys", since = "1.10.0")]
     pub fn key(&self) -> &K {
         self.elem.read().0
     }
 
     /// Take the ownership of the key and value from the map.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(map_entry_recover_keys)]
+    ///
+    /// use std::collections::HashMap;
+    /// use std::collections::hash_map::Entry;
+    ///
+    /// let mut map: HashMap<&str, usize> = HashMap::new();
+    /// map.entry("poneyland").or_insert(12);
```

Comment:
> I mean something like let map: HashMap<&str, usize> = [("key", 42)].iter().cloned().collect();

I think the current solution is cleaner than using `collect`.

> You also should probably just use u32/u64, and not usize because of variability of usize between targets.

How does the variability of usize affect the example?